tests/locals.tftest.hcl

```diff
@@ -0,0 +1,410 @@
+  }
+}
+
+run "test_groups_map_transformation" {
+  command = plan
+
+  assert {
+    condition = length(local.groups_map) == 2
+    error_message = "groups_map should contain 2 groups"
+  }
+
+  assert {
+    condition = local.groups_map["Administrators"].description == "Admin group"
+    error_message = "Administrators group should have correct description"
+  }
+
+  assert {
+    condition = length(local.groups_map["Developers"].members) == 2
+    error_message = "Developers group should have 2 members"
+  }
+}
+
+run "test_permission_sets_map_transformation" {
+  command = plan
+
+  assert {
+    condition = length(local.permission_sets_map) == 3
+    error_message = "permission_sets_map should contain 3 permission sets"
+  }
+
+  assert {
+    condition = local.permission_sets_map["AdminAccess"].session_duration == "PT8H"
+    error_message = "AdminAccess should have 8 hour session duration"
+  }
+
+  assert {
+    condition = length(local.permission_sets_with_inline_policy_map) == 1
+    error_message = "Only ReadOnly should have inline policy"
+  }
+
+  assert {
+    condition = local.permission_sets_with_inline_policy_map["ReadOnly"].inline_policy != null
+    error_message = "ReadOnly should have non-null inline policy"
+  }
+}
+
+run "test_group_memberships_flattening" {
+  command = plan
+
+  assert {
+    condition = length(local.group_memberships) == 4
+    error_message = "Should have 4 total group memberships"
+  }
+
+  assert {
+    condition = length(local.group_memberships_map) == 4
+    error_message = "group_memberships_map should have 4 entries"
+  }
+
+  # Test specific membership structures
+  assert {
+    condition = local.group_memberships[0].group_name == "Administrators" && local.group_memberships[0].user_name == "john.doe"
+    error_message = "First membership should be Administrators-john.doe"
+  }
+
+  assert {
+    condition = local.group_memberships[1].group_name == "Administrators" && local.group_memberships[1].user_name == "jane.smith"
+    error_message = "Second membership should be Administrators-jane.smith"
+  }
+
+  assert {
+    condition = local.group_memberships[2].group_name == "Developers" && local.group_memberships[2].user_name == "jane.smith"
+    error_message = "Third membership should be Developers-jane.smith"
+  }
+
+  assert {
+    condition = local.group_memberships[3].group_name == "Developers" && local.group_memberships[3].user_name == "bob.jones"
+    error_message = "Fourth membership should be Developers-bob.jones"
+  }
+
+  # Test specific map entries
+  assert {
+    condition = local.group_memberships_map["Administrators-john.doe"].group_name == "Administrators"
+    error_message = "Administrators-john.doe map entry should have correct group_name"
+  }
+
+  assert {
+    condition = local.group_memberships_map["Administrators-john.doe"].user_name == "john.doe"
+    error_message = "Administrators-john.doe map entry should have correct user_name"
+  }
+
+  assert {
```

Comment:
I feel like I'm not seeing the difference in tests on lines 185-194 and 201-210 ? I see that they are both testing that the `group_name` and `user_name` are correct, but I don't see differences beyond that. Can you help me see more of what's happening in the tests?